packages/visualizations/src/components/Table/Pagination/Pages.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+    import Button from './Button.svelte';
+    import DoubleLeft from './DoubleLeft.svelte';
+    import DoubleRight from './DoubleRight.svelte';
+    import SingleLeft from './SingleLeft.svelte';
+    import SingleRight from './SingleRight.svelte';
+
+    export let totalPages: number;
+    export let current: number;
+    export let setPage: (page: number) => void;
+
+    let pages = [current - 1, current, current + 1];
+    $: if (totalPages === 2) {
+        pages = [1, 2];
+    } else if (totalPages === 1) {
+        pages = [1];
+    } else if (current - 1 < 1) {
+        pages = [current, current + 1, current + 2];
+    } else if (current + 1 > totalPages) {
+        pages = [current - 2, current - 1, current];
+    } else {
+        pages = [current - 1, current, current + 1];
+    }
```

Comment:
It's hard to read, IMO we can improve it:

If totalPages is equal to 3 are less: pages = `Array.from({length: totalPages}, (_, i) => i + 1)`
if current === 1: `pages = [current, current + 1 , current + 2];`
if current === totalPages: `pages = [current - 2, current - 1, current];`
All other cases: `pages = [current - 1, current, current + 1];`